### ISSUE TITLE
`jinja-variable-lower-case` wrongly detects printed strings as variables

### ISSUE THREAD

**Issue by tacerus** [aristanetworks/j2lint#85]
Hi,

if printing strings inside of a `{{` statement, the tool complains about `jinja-variable-lower-case`:

For example:

```
{{ ' trustGuestRxFilters=\'yes\'' if iftype == 'direct' else '' }}
```

Results in:

```
All variables should use lower case: '{{ variable }}': trustGuestRxFilters (jinja-variable-lower-case)
```

Albeit `trustGuestRxFilters` not being a variable, but a printed string. The only variable in this line is `iftype`, which is lower case.

I think it is wrongly assuming `{{` to only contain variable names?

**Comment by gmuloc** [comment]
Hello - thanks for opening the issue - the rule is using the following regex and is indeed expecting that it contains only a variable name

```
regex = re.compile(r"([a-zA-Z0-9-_\"']*[A-Z][a-zA-Z0-9-_\"']*)")
```
https://github.com/aristanetworks/j2lint/blob/devel/j2lint/rules/jinja_variable_name_case_rule.py#L27

The rule does not support inline ifs (https://jinja.palletsprojects.com/en/3.1.x/templates/#if-expression) as it was never required until now (read: our main project using j2lint never uses them). 
The support for this may take some time to implement and will require a smarter `get_jinja_variables` function: https://github.com/aristanetworks/j2lint/blob/devel/j2lint/utils.py#L229

In the meantime, you can always add an ignore statement for j2lint for this specific rule at the end of this line.

**Comment by tacerus** [comment]
Hi,

thanks for the input. I understand it's not directly relevant to the intended implementation of j2lint.

Based on your pointers, I came up with this for `get_jinja_variables`:

https://github.com/tacerus/j2lint/commit/98da74a7addb85d093e211daaa52c7ed20d0367a

Whilst it works for my example (extracts `iftype`), it likely needs more consideration for different kinds of if-expressions as they might not always be just comparisons, and could theoretically be split over multiple lines.
Seems my request is indeed a bit tricky! I will ignore the rule for now, but if this could stay a feature request it would be great.

**Comment by DeadNews** [comment]
A similar situation with [escaping](https://jinja.palletsprojects.com/en/3.1.x/templates/#escaping):

```jinja2
template: '{{ '{{ ToLower .Value }}' }}'
```

```console
$ j2lint 

All variables should use lower 
│   case: '{{ variable }}': ToLower (jinja-variable-lower-case)
```

**Comment by cdtomkins** [comment]
This will be closed by https://github.com/aristanetworks/j2lint/pull/132
